pipeline/backend/kubernetes/image_pull_secret_test.go

```diff
@@ -0,0 +1,97 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/kinbiko/jsonassert"
+	"github.com/stretchr/testify/assert"
+	"go.woodpecker-ci.org/woodpecker/v2/pipeline/backend/types"
+)
+
+func TestNoAuthNoSecret(t *testing.T) {
+	assert.False(t, needsImagePullSecret(&types.Step{}))
+}
+
+func TestNoPasswordNoSecret(t *testing.T) {
+	assert.False(t, needsImagePullSecret(&types.Step{
+		AuthConfig: types.Auth{Username: "foo"},
+	}))
+}
+
+func TestNoUsernameNoSecret(t *testing.T) {
+	assert.False(t, needsImagePullSecret(&types.Step{
+		AuthConfig: types.Auth{Password: "foo"},
+	}))
+}
+
+func TestUsernameAndPasswordNeedsSecret(t *testing.T) {
+	assert.True(t, needsImagePullSecret(&types.Step{
+		AuthConfig: types.Auth{Username: "foo", Password: "bar"},
+	}))
+}
+
+func TestImagePullSecret(t *testing.T) {
+	expected := `{
+		"metadata": {
+			"name": "wp-01he8bebctabr3kgk0qj36d2me-0",
+			"namespace": "woodpecker",
+			"creationTimestamp": null,
+			"labels": {
+				"app": "test",
+				"part-of": "woodpecker-ci",
+				"step": "go-test"
+			},
+			"annotations": {
+				"apps.kubernetes.io/pod-index": "0",
+				"kubernetes.io/limit-ranger": "LimitRanger plugin set: cpu, memory request and limit for container"
+			}
+		},
+  	"type": "kubernetes.io/dockerconfigjson",
+	  "data": {
+			".dockerconfigjson": "eyJhdXRocyI6eyJkb2NrZXIuaW8iOnsidXNlcm5hbWUiOiJmb28iLCJwYXNzd29yZCI6ImJhciJ9fX0="
+	  }
+	}`
+
+	pullSecret, err := mkImagePullSecret(&types.Step{
```

Comment:
You don't need a half of options to test secrets.

Reply:
Removed all non-relevant fields